src/tests/users.test.ts

```diff
@@ -0,0 +1,290 @@
+  const formData: UserType = {
+    username: 'test1',
+    firstName: 'test1',
+    lastName: 'test1',
+    password: 'test1',
+  };
+  await createUserPage.fillUserForm(formData);
+  await createUserPage.saveButton.click();
+  await expect(editUserPage.userTitle).toContainText('test1 test1');
+  await editUserPage.activateUser.click();
+  await editUserPage.saveButton.click();
+  await editUserPage.authorizationTab.click();
+  await editUserPage.defaultRoleSelect.click();
+  await editUserPage.getUserRole('Manager').click();
+  await editUserPage.defaultLocation.click();
+  const location = await mainLocation.getLocation();
+  await editUserPage.getDefaultLocation(location.name).click();
+  await editUserPage.autoLogin.click();
+  await editUserPage.saveButton.click();
+  await navbar.profileButton.click();
+  await navbar.logoutButton.click();
+  await loginPage.fillLoginForm(formData.username, formData.password);
+  await loginPage.loginButton.click();
+  await expect(page.getByText('My Dashboard')).toBeVisible();
+  await navbar.profileButton.click();
+  await navbar.logoutButton.click();
+  await loginPage.fillLoginForm(
+    AppConfig.instance.users.main.username,
+    AppConfig.instance.users.main.password
+  );
+  await loginPage.loginButton.click();
+  await locationChooser.getLocation(location.name).click();
+  await navbar.configurationButton.click();
+  await navbar.getNavItem('Users').click();
+  await userListPage.searchByNameField.fill('test1');
+  await userListPage.findButton.click();
+  await userListPage.getUserToEdit('test1').click();
+  await editUserPage.actionButton.click();
+  page.on('dialog', (dialog) => dialog.accept());
+  await editUserPage.deleteUserButton.click();
+  await userListPage.searchByNameField.fill('test1');
+  await userListPage.findButton.click();
+  await expect(userListPage.userListTable).toContainText('No users returned');
+});
+
+test('Create and impersonate user', async ({
+  navbar,
+  userListPage,
+  createUserPage,
+  editUserPage,
+  page,
+}) => {
+  test.setTimeout(3 * 60 * 1000);
+  await navbar.configurationButton.click();
+  await navbar.getNavItem('Users').click();
+  await userListPage.createUserButton.click();
+  const formData: UserType = {
+    username: 'test2',
+    firstName: 'test2',
+    lastName: 'test2',
+    password: 'test2',
+  };
+  await createUserPage.fillUserForm(formData);
+  await createUserPage.saveButton.click();
+  await expect(editUserPage.userTitle).toContainText('test2 test2');
+  await editUserPage.activateUser.click();
+  await editUserPage.saveButton.click();
+  await editUserPage.authorizationTab.click();
+  await editUserPage.defaultRoleSelect.click();
+  await editUserPage.getUserRole('Manager').click();
+  await editUserPage.saveButton.click();
+  await editUserPage.impersonateButton.click();
+  const newPage = await page.waitForEvent('popup');
+  await expect(newPage.getByRole('alert')).toContainText(
+    'You are impersonating user test2'
+  );
```

Comment:
create a POM for impersonate banner and move this assertion into `isLoaded()`
the just call this method in the test
```ts
await impersonateBanner.isLoaded(formData.username)

```